service/test/BatchServerTest.cpp

```diff
@@ -426,3 +730,441 @@ TEST_F(BatchServerTest, fork_with_setup)
+    int forked_pid = 0;
+    GEOPM_EXPECT_THROW_MESSAGE((forked_pid = m_batch_server_pid->fork_with_setup(setup, run)),
+                               GEOPM_ERROR_RUNTIME,
+                               "BatchServerImp: Receivied unexpected message from batch server at startup: \"" "113" "\"");
+    waitpid(forked_pid, NULL, 0);
+}
+
+/**
+ * @class CerrRedirect
+ *
+ * @details This class facilitates redirect the stderr to an internal buffer for recording purposes.
+ */
+class CerrRedirect {
+    public:
+        /**
+         * @brief The redirect is NOT active when the object is created.
+         */
+        CerrRedirect()
+        {
+            cerr_backup = nullptr;
+            is_active = false;
+        }
+
+        /**
+         * @brief Restore the backup streambuf if it was not restored already.
+         */
+        ~CerrRedirect()
+        {
+            if (cerr_backup) {
+                std::cerr.rdbuf(cerr_backup);
+            }
+            is_active = false;
+        }
+
+        /**
+         * @brief Call this function first. Activates the redirect.
+         *
+         * @details After calling this function, all std::cerr will be redirected to the internal stringstream.
+         *          This function saves the backup streambuf of std::cerr, and resets it to the internal stringstream buffer.
+         */
+        void open_redirect(void)
+        {
+            if (!is_active) {
+                // Flush the remaining stderr into the terminal.
+                std::cerr << std::flush;
+                // Clear the stringstream
+                buffer.clear();
+                buffer.str(std::string());
+                // Set the streambuf of the std::cerr equivalent to the streambuf of the stringstream.
+                // Return the former streambuf of the std::cerr prior to the change.
+                cerr_backup = std::cerr.rdbuf(buffer.rdbuf());
+                is_active = true;
+            }
+        }
+
+        /**
+         * @brief This function deactivates the redirect. It should be called after open_redirect().
+         *
+         * @details This function restores the backup streambuf associated with the stderr, and closes the stringstream redirect.
+         *
+         * @return The internal string of the stringstream, which recorded the redirected output.
+         */
+        std::string close_redirect(void)
+        {
+            if (is_active) {
+                // Flush the remaining stderr into the streambuf.
+                std::cerr << std::flush;
+                // Restore the streambuf of the std::cerr to it's original state.
+                std::cerr.rdbuf(cerr_backup);
+                is_active = false;
+
+                return buffer.str();
+            }
+            else {
+                return std::string();
+            }
+        }
+    private:
+        // records the redirected stderr into it's own streambuf
+        std::stringstream buffer;
```

Comment:
```suggestion
        std::stringstream m_buffer;
```
All class members should start with "m_":